### PR TITLE
Illumos systems/Solaris support.

### PR DESCRIPTION
**Pull request by devnexen**
MAP_NORESERVE flag usage for memory over commit permitted only
 in the mmap context.

### CODE REVIEW

**Comment by SchrodingerZhu** [comment]
will this library be compiled by Solaris Studio or GCC?

**Comment by devnexen** [comment]
tried with clang and gcc. would sun studio handles c++17 tough ?

File: src/pal/pal_solaris.h [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#if defined(__sun)
+// clang-format off
```
Comment (davidchisnall) [inline src/pal/pal_solaris.h:4]: Why is this needed?

Comment (devnexen) [reply src/pal/pal_solaris.h:4]: ah ... clang-format had the tendency to reorder the inclusions.

Comment (davidchisnall) [reply src/pal/pal_solaris.h:4]: It shouldn't reorder between blocks - isn't the blank line between them enough?

Comment (devnexen) [reply src/pal/pal_solaris.h:4]: ah ... still reordering it.

Comment (mjp41) [reply src/pal/pal_solaris.h:4]: I would sooner you add 
```
#include <strings.h>
```
into the Pal posix header.  That is calling `bzero`, which is coming from `strings.h`.  I guess other platforms have exposed it without requiring the inclusion.

Comment (devnexen) [reply src/pal/pal_solaris.h:4]: I considered this but was not sure all posix systems has this. We can always try.
